content/200-concepts/100-components/03-prisma-migrate/010-mental-model.mdx

```diff
@@ -0,0 +1,163 @@
+
+Migrations provide _state management_ which helps you to track the state of the database.
+
+Migrations also allow you to replicate the state of a database at a specific point in time which is useful when collaborating with other members of the team, e.g. switching between different branches.
+
+For further information on database migrations, see the [Prisma Data Guide](https://www.prisma.io/dataguide/types/relational/what-are-database-migrations).
+
+## What is Prisma Migrate?
+
+Prisma Migrate is a database migration tool that supports _model/ entity-first_ migration and _database first_ migration patterns to manage database schemas in your local environment and in production.
+
+The workflow when using Prisma Migrate in your project would be iterative and look like this:
+
+- Feature branch (locally)
+  1. Define your Prisma schema – the source of truth
+  1. Use one of the [Prisma CLI](/reference/api-reference/command-reference) commands to sync your Prisma schema with the database schema
+- Feature pull request (preview/ staging environment)
+  1. Push your changes to the feature pull request
+  1. Sync your Prisma schema and migration history with your preview database using `prisma migrate deploy`
+- Main branch (Production)
+  1. Merge your application code from the feature branch to your main branch
+  1. Sync your Prisma schema and migration history with your preview database using `prisma migrate deploy`
+
+![Prisma Migrate workflow](https://user-images.githubusercontent.com/33921841/194152691-1aa0a8cf-1508-488b-b734-cd6d05abd54f.png)
+
+## How Prisma Migrate tracks the migration state
+
+Prisma Migrate uses the following pieces of state to track the state of your database schema:
+
+- **Prisma schema**: your source of truth that defines the structure of the database schema.
+- **Migrations history**: SQL files in your `prisma/migrations` folder representing the history of changes made to your database schema.
+- **Migrations table**: `prisma_migrations` table in the database that stores metadata for migrations that have been applied to the database.
+- **Database schema**: the state of the database.
+
+![Prisma Migrate "state management"](https://user-images.githubusercontent.com/33921841/189885586-0e0d7852-8ac7-4842-a6cb-bd158c20d28e.png)
+
+## Assumptions when working with Prisma Migrate
+
+- Ideally, you should use different databases for different environments. For example, you might have a separate database for development, preview, and production environments.
+- The databases you use in development environments are disposable — you can easily create, use, and delete databases on demand.
+- The database configuration used in different environments should be similar to ensure it is reproducible.
+- The Prisma schema serves as the source of truth — describing the shape of your [database schema](https://www.prisma.io/dataguide/intro/database-glossary#schema).
```

Comment:
by the way, this contradicts that Prisma Migrate supports the DB-first approach. by definition, in the DB-first approach SQL is the source of truth, so this contradicts this "assumption when working with Prisma Migrate".

Reply:
I've removed database-first migration for the definition. That should fix it 🙃